packages/core/src/lib/implementation/read-rc-file.unit.test.ts

```diff
@@ -0,0 +1,96 @@
+import { vol } from 'memfs';
+import { describe, expect, vi } from 'vitest';
+import { CONFIG_FILE_NAME, CoreConfig } from '@code-pushup/models';
+import { MEMFS_VOLUME } from '@code-pushup/testing-utils';
+import { autoloadRc } from './read-rc-file';
+
+// Mock bundleRequire inside importEsmModule used for fetching config
+vi.mock('bundle-require', async () => {
+  const { CORE_CONFIG_MOCK }: Record<string, CoreConfig> =
+    await vi.importActual('@code-pushup/testing-utils');
+
+  return {
+    bundleRequire: vi
+      .fn()
+      .mockImplementation((options: { filepath: string }) => {
+        const extension =
+          options.filepath?.split('.').at(-1) || 'no-extension-found';
+        return {
+          mod: {
+            default: {
+              ...CORE_CONFIG_MOCK,
+              upload: {
+                ...CORE_CONFIG_MOCK?.upload,
+                project: extension,
+              },
+            },
+          },
+        };
+      }),
+  };
+});
+
+describe('autoloadRc', () => {
+  it('prioritise a .ts configuration file', async () => {
+    vol.fromJSON(
+      {
+        // this is just here to satisfy the file system check. the file ise served over a mock in bundleRequire
+        [`${CONFIG_FILE_NAME}.js`]: '',
+        [`${CONFIG_FILE_NAME}.mjs`]: '',
+        [`${CONFIG_FILE_NAME}.ts`]: '',
+      },
+      MEMFS_VOLUME,
+    );
+
+    await expect(autoloadRc()).resolves.toEqual(
+      expect.objectContaining({
+        upload: expect.objectContaining({
+          project: 'ts',
```

Comment:
I think it would be better to move the comments here to explain where the value is coming from (also care, there is a 'ise' typo in all three comments right now).

Reply:
added a comment

Reply:
is this a blocker for this PR